examples/rl/rllib/rllib.py

```diff
@@ -103,13 +108,78 @@ def main(
     num_workers,
     resume_training,
     result_dir,
-    checkpoint_num,
+    checkpoint_freq: int,
+    checkpoint_num: Optional[int],
+    log_level: Literal["DEBUG", "INFO", "WARN", "ERROR"],
     save_model_path,
 ):
     assert train_batch_size > 0, f"{train_batch_size.__name__} cannot be less than 1."
-    if rollout_fragment_length > train_batch_size:
+    if (
+        isinstance(rollout_fragment_length, int)
+        and rollout_fragment_length > train_batch_size
+    ):
         rollout_fragment_length = train_batch_size
 
+    rllib_policies = {
+        f"AGENT-{i}": (
+            None,
+            rllib_agent["observation_space"],
+            rllib_agent["action_space"],
+            {"model": {"custom_model": TrainingModel.NAME}},
+        )
+        for i in range(num_agents)
+    }
+
+    smarts.core.seed(seed)
+    algo_config = (
+        PGConfig()
```

Comment:
The typed config appears to help with generating the example from the code side of things. I somewhat wonder about working with it from a different approach.

It also appears possible that as long as we register the algorithm it should be possible to use the [`rllib train`](https://docs.ray.io/en/latest/rllib/rllib-cli.html) to run our examples using a custom configuration.